site/documentation/content/user-guide/coap-adapter.md

```diff
@@ -0,0 +1,231 @@
++++
+title = "CoAP Adapter"
+weight = 210
++++
+
+The CoAP protocol adapter exposes [CoAP](https://tools.ietf.org/html/rfc7252) based endpoints for Eclipse Hono&trade;'s south bound Telemetry, Event and Command & Control APIs.
+<!--more-->
+
+## Device Authentication
+
+The CoAP adapter by default requires clients (devices or gateway components) to authenticate during connection establishment. The adapter supports currently only [PSK](https://tools.ietf.org/html/rfc4279) as part of a DTLS handshake for that purpose. The other variants mentioned in [Securing CoAP](https://tools.ietf.org/html/rfc7252#section-9) may follow in the future.
+
+The adapter tries to authenticate the device using this mechanisms
+
+### PSK
+
+The *identity* provided in the ClientKeyExchange must have the form *auth-id@tenant*, e.g. `sensor1@DEFAULT_TENANT`. The adapter performs the handshake using the credentials `configured Credentials service - (documentation pending)` has one record for the client. The adapter uses the Credentials API's *get* operation to retrieve the credentials on record with the *tenant* and *auth-id* provided by the device in the *identity* and `psk` as the *type* of secret as query parameters.
+
+The examples below refer to devices `4711` and `gw-1` of tenant `DEFAULT_TENANT` using *auth-ids* `sensor1` and `gw1` and corresponding passwords. The example deployment as described in the [Deployment Guides]({{< relref "deployment" >}}) comes pre-configured with the corresponding entities in its device registry component.
+Please refer to the [Credentials API]({{< relref "/api/credentials#standard-credential-types" >}}) for details regarding the different types of secrets.
+
+**NB** There is a subtle difference between the *device identifier* (*device-id*) and the *auth-id* a device uses for authentication. See [Device Identity]({{< relref "/concepts/device-identity.md" >}}) for a discussion of the concepts.
+
+## Message Limits
+
+Before accepting any telemetry or event or command messages, the CoAP adapter verifies that the configured [message limit] ({{< relref "/concepts/resource-limits.md" >}}) is not exceeded. If the limit is exceeded then the incoming message is discarded with the status code `429 Too Many Requests`. 
```

Comment:
is this actually the case?

Reply:
`checkMessageLimit` is called, but I didn't check, if it really works.

@kaniyan 
Anything else, which must be called?

Reply:
@boaks The adapters use only `checkMessageLimit` from the base class to verify message limit.

Reply:
However, my understanding is that we (currently) do not report established (DTLS) connections so we are probably only checking for overall message data volume. If so, we should document this fact here ...

Reply:
You mean, add that there is currently no forced "connection limit"?

Reply:
yes, I would propose to do so.

Reply:
Ok